File: app/src/main/java/com/penseapp/acaocontabilidade/login/presenter/LoginPresenter.java
package com.penseapp.acaocontabilidade.login.presenter;

/**
 * Created by Filip on 23/02/2016.
 */
public interface LoginPresenter {
    void signUp(String name, String email, String type, String password);
    void onSignUpFailure();
    void onSignUpSuccess(String email, String uid);

    void login(String email, String password);
    void onLoginFailure();
    void onLoginSuccess(String user, String uid);

    void resetPassword(String email);
    void onPasswordResetSuccess();
    void onPasswordResetFailure();

    void getCurrentUser();
    void onUserLoggedIn();
    void onUserLoggedOut();

    void addAuthStateListener();
    void removeAuthStateListener();
}

File: app/src/main/java/com/penseapp/acaocontabilidade/login/presenter/LoginPresenter.kt
package com.penseapp.acaocontabilidade.login.presenter

/**
 * Created by Filip on 23/02/2016.
 */
interface LoginPresenter {

    fun signUp(name: String, company: String, email: String, type: String, password: String)

    fun onSignUpFailure()

    fun onSignUpSuccess(email: String, uid: String)


    fun login(email: String, password: String)

    fun onLoginFailure()

    fun onLoginSuccess(user: String, uid: String)


    fun resetPassword(email: String)

    fun onPasswordResetSuccess()

    fun onPasswordResetFailure()


    fun getCurrentUser()

    fun onUserLoggedIn()

    fun onUserLoggedOut()


    fun addAuthStateListener()

    fun removeAuthStateListener()
}
